jax/experimental/optimizers.py

```diff
@@ -423,3 +423,51 @@ def clip_grads(grad_tree, max_norm):
   norm = l2_norm(grad_tree)
   normalize = lambda g: np.where(norm < max_norm, g, g * (max_norm / norm))
   return tree_map(normalize, grad_tree)
+
+
+### serialization utilities
+
+class JoinPoint(object):
+  """Marks the boundary between model params and optimizer-only params."""
```

Comment:
How about "Marks the boundary between two joined pytrees."? It's not exactly that it marks the boundary between model params and optimizer-only params, since the model params live under the JoinPoint too. It does mark the boundary between the tree defined by the model parameters and the subtree defined by the optimizer state. But for such a simple object I think it'd be better to have a more general docstring.

Reply:
Done.